internal/providersdkv2/provider.go

```diff
@@ -210,10 +211,23 @@ func getProjectFromCredentials(ctx context.Context, client *clients.Client) (pro
 		return nil, diags
 	}
 	orgLen := len(listOrgResp.Payload.Organizations)
-	if orgLen != 1 {
-		diags = append(diags, diag.Errorf("unexpected number of organizations: expected 1, actual: %v", orgLen)...)
+	if orgLen == 0 {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "The configured credentials do not have access to any organization.",
+			Detail:   "Please assign at least one organization to the configured credentials to use this provider.",
```

Comment:
Unfortunately, there is no place to configure the organization id. https://github.com/hashicorp/terraform-provider-hcp/blob/44d4356be6c4daf126b975ab2246b2990b85b8a2/internal/providersdkv2/provider.go#L62-L115 It only takes project id. We will need to update it to take org id and run through different workflows (with user creds, sp creds with diff level access) to see it works. This should fix #677.

Reply:
I see, I was confused because there was a field in the config that made me think that it was possible https://github.com/hashicorp/terraform-provider-hcp/blob/62e7c8949bc35095b0d99af8f0dd72fdaa7f2e3c/internal/clients/client.go#L101

Reply:
@manish-hashicorp Do you have any recommendations on how to rephrase this error?

Reply:
Not really until we add support for configuring organization which would be the real fix. Looks like that config is set to be the parent of configured or fetched project id.